Download page - Add download suggestion page for Lime3DS releases

SrShadowy opened the pull request:
**Good day everyone!**
I hope you're all doing very well!

A friend of mine was using the late citra and I recommended Lime3DS and he asked me *can I download it from this site ?* 😢 he's just a regular user, who's had bad experiences with Github, so I had this idea: why not create a download page that uses the GitHub API and some markdown for the changelog? 🤔 


Instead of just redirecting users directly to the version downloads on GitHub (which works, but hey, we can do better), this update introduces a small suggestion page 🎉🎉🎉🎉🎉. It shows the latest Lime3DS releases, allows users to browse the changelog and, of course, download the assets directly from the site, oooh yeah much easier. Oh, and there's even a button to view previous versions, if you're missing them! 🎉 yeah, I didn't forget, the Github api delivers everything hands down.

*I hope I didn't write anything wrong, forgive my English*

Here's the magic of the APIs I used:
- **GitHub API** to get the versions
- **Marked.js** to render the markdown: https://github.com/markedjs/marked  
  (You can get it from the CDN here: https://www.cdnpkg.com/marked/file/marked.min.js/)

And, of course, the official GitHub documents for reference: https://docs.github.com/en/

![Captura de tela 2024-09-17 182139](https://github.com/user-attachments/assets/de31ae08-d276-4654-bee8-cd381078efb4)

SrShadowy committed:
Download page

Why not download page using markdown and api of github?

SrShadowy committed:
Add footer and div content

SrShadowy committed:
add flatpak buttons

ghost commented:
Ah sorry for missing this i am not watching this repo

@OpenSauce04 this is sick ngl

kleidis commented:
Although the colors can be tweaked and maybe icons should be used. I like this as a base tho

ghost commented:
I'm not much of a web developer, but this code looks clean to me 
I'll try to make it prettier on some days after it's merged

OpenSauce04 commented on pages/download/index.html:
```diff
@@ -0,0 +1,93 @@
+                        </a>
+                    </li>
+                </ul>
+                <h2>Changelog:</h2>
+                <h3>${release.name}</h3>
+                <div>
+                    ${marked(release.body || "No release notes available.")}
+                </div>
```
I would drop the release notes from this page, as it makes the page layout a bit awkward and lengthy. If a user wishes to view the changelog, they can do so from our GitHub page.

kleidis replied:
Perhaps hiding it by default and adding a drop-down button wouldn't be a bad idea?
I do agree, showing it can be much depending on notes

SrShadowy replied:
hiding by default can be a good idea

OpenSauce04 commented on resources/stylesheets/download.css:
```diff
@@ -0,0 +1,51 @@
+    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
+}
+
+.windows {
+    background-color: #0078D7;
+}
+
+.linux {
+    background-color: #FCC624;
+    color: #333;
+}
+
+.mac {
+    background-color: #A4C8E1;
+}
+
+.android {
+    background-color: #A4C639;
+}
```
These colour choices are too bright, and have poor contrast with the white text, making for a poor visual experience when using the website.

Coincidentally, I think my dark mode extension which I left on by accident while testing this page has a good selection of colours, so try to go for something more similar to this (but not necessarily the same):

<img src=https://github.com/user-attachments/assets/a5d7a49a-becb-473c-9b54-29dc7a940152 width=300px/>

Additionally, the choices of colour for Mac and Linux make little sense. Perhaps try black and white for Linux and MacOS respectively? My logic is following Tux and the Apple logo's conventional colour palettes. I am open to other suggestions if you have them.

kleidis replied:
We really can do better than that

OpenSauce04 replied:
> We really can do better than that

What are you referring to?

kleidis replied:
css deisgn

kleidis replied:
Clarification:

Having this many big buttons isn't really ideal

We could either detect the platform the wen browser is on, make that the main button, and add smaller elements for the rest of the platforms or just combine the UI elements in a way that looks better if we don't wanna go through tha trouble of adding the OS detective code

OpenSauce04 replied:
I was going to get to that, but I wanted to get this out of the way first to avoid conflicting change requests

SrShadowy replied:
So.... maybe making a design and putting an icon for each system would be a viable option, the point is that I put the color separation to speed up loading 
Also, @kleidis had a great idea about detecting the system in js and it's not difficult, but the size of the button depends on the fact that Windows has two options and so does Android.
Would you show both and put one by default? apk and msvc?

kleidis replied:
Idk what OpenSauce likes for Wibdows default to he 
For now, put MSVC

Android should definitely be the apk

OpenSauce04 replied:
If this kind of download functionality is to be added to the website, all relevant downloads must be available.

kleidis replied:
> If this kind of download functionality is to be added to the website, all relevant downloads must be available

You mean that all buttons should be visible no matter what platform?

SrShadowy replied:
a made changes on windows for now look this

is good?

![msedge_NfjfNVPv3v](https://github.com/user-attachments/assets/e17bddb5-fd6e-46f7-ae75-f8d57df89576)

I don't know about css of buttons yet

kleidis replied:
I'm just throwing this as a snippet

Could we just do it like this?

![Screenshot_20240924_214624_Kiwi Browser.jpg](https://github.com/user-attachments/assets/efc2e4dc-066d-4a73-a946-a7e967590bb8)

SrShadowy replied:
how about types? we have 2 buttons for windows and android, put side by side?

SrShadowy replied:
here another way I think before

![image](https://github.com/user-attachments/assets/9283da1e-bf54-4560-894e-6680c7d95373)

-


![firefox_mXeKVEwiqu](https://github.com/user-attachments/assets/a19cbdea-e271-47b0-89a4-ff4361ae1f12)

SrShadowy replied:
![image](https://github.com/user-attachments/assets/72cd471d-aac4-4d9b-aedc-f406500e97ed)
Buttons like this?

kleidis replied:
Waitibg for @OpebSauce04 response

OpenSauce04 replied:
> here another way I think before

The second of those options looks good. Could do with a little extra styling to make it feel more integrated with the website but that's something I can handle.

Why are the labels still formatted like filenames by the way? That seems like a pretty strange decision. Just have `Windows (MSVC)`, `Linux (AppImage)`, etc.

Also, the AAB doesn't need to be on the website. It doesn't really have a purpose to an end user.

Push the code for that version of the download page and I'll take it from there.

ayrudev replied:
> how about types? we have 2 buttons for windows and android, put side by side?

i think a dropdown might be the better option? maybe

OpenSauce04 commented on pages/download/index.html:
```diff
@@ -0,0 +1,93 @@
+                .join("");
+
+            releaseDiv.innerHTML = `
+                <h4>Download:</h4>
```
```suggestion
```

OpenSauce04 commented on pages/download/index.html:
```diff
@@ -0,0 +1,93 @@
+    <script>HTMLImporter.import("/resources/templates/generic-nav.html");</script>
+
+    <div id="content">
+        <h1>Latest Updates on GitHub - Lime3DS</h1>
```
```suggestion
        <h1>Latest Lime3DS Downloads</h1>
```

OpenSauce04 commented on pages/download/index.html:
```diff
@@ -0,0 +1,93 @@
+                <ul>
+                    ${assetList}
+                    <li>
+                        <p>We are also on Flathub:</p>
```
```suggestion
```

OpenSauce04 replied:
Stating that we are on Flathub next to the big button that says "Flathub" is redundant

ayrudev replied:
> Stating that we are on Flathub next to the big button that says "Flathub" is redundant

i mean... it's the same as on https://github.com/Lime3DS/Lime3DS?tab=readme-ov-file#linux

![image](https://github.com/user-attachments/assets/c66f21bf-aa19-4d09-b052-8bf10cbe69f5)

OpenSauce04 replied:
The difference is that our readme has descriptions for all of our downloads, whereas here the Flathub download is the only one with such text.

SrShadowy committed:
Update pages/download/index.html

Co-authored-by: OpenSauce <opensauce04@gmail.com>

SrShadowy committed:
Update pages/download/index.html

Co-authored-by: OpenSauce <opensauce04@gmail.com>

SrShadowy committed:
Buttons and layout

This code is not CLEAR! please read with attention

on const testing for design test with block and image for icons system

if u turn to false, back to original wihout greater changes on buttons.

SrShadowy commented:
any new suggestion?

OpenSauce04 commented:
Sorry, a little busy at the moment, may take a couple of days for me to get back to you on this

dumor1990 commented:
Add hacks fps for android user.

ayrudev commented:
maybe this should be updated to show the google play link

dumor1990 commented:
Nice saw it can you add it in android 11 TV Hisense lot emulator in Google
Play tv

On Sun, Oct 20, 2024, 6:25 PM imnotpolar ***@***.***> wrote:

> maybe this should be updated to show the google play link
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Lime3DS/lime3ds.github.io/pull/27#issuecomment-2425258630>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQM4763LB576XAV4T6EHKO3Z4QUWRAVCNFSM6AAAAABOKT3A76VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIMRVGI2TQNRTGA>
> .
> You are receiving this because you commented.Message ID: <Lime3DS/lime3ds.
> ***@***.***>
>

Moriko-Nanami commented:
I'd love to download the Lime, however, I can't find the link or button for doing so.  If I could I would because Citra no longer works for me and I haven't the slightest clue why.

ayrudev commented:
@Moriko-Nanami please check out the [Azahar Emulator](https://github.com/azahar-emu/azahar), Lime3DS was discontinued and Azahar is it's successor.
